src/Utopia/Messaging/Adapter/Email/SMTP.php

```diff
@@ -76,6 +76,38 @@ protected function process(EmailMessage $message): array
             $mail->addAddress($to);
         }
 
+        if (!empty($message->getCC())) {
+            foreach ($message->getCC() as $cc) {
+                $mail->addCC($cc['email'], $cc['name']);
+            }
+        }
+
+        if (!empty($message->getBCC())) {
+            foreach ($message->getBCC() as $bcc) {
+                $mail->addBCC($bcc['email'], $bcc['name']);
+            }
+        }
+
+        if (!empty($message->getAttachments())) {
```

Comment:
Anyway for us to abstract some of this in the Email base class, I see similar code in 3 adapters

Reply:
Can't think of a way to do it in the email adapter without needing to call `parent::process` from the child classes. We could add the size checks in the base `Adapter`, but it feels wrong to add email specific code in here